stanify/calibrator/draws_data_mapper.py

```diff
@@ -0,0 +1,177 @@
+    """
+
+    model = transform_input(vensim, setting, precision, numeric, prior, idata_kwargs)
+    model.stanify_ode_function()
+    # which info to get from which context
+
+    data_dict = {
+        **model.numeric,
+        **precision
+    }
+    idata_orig = draws2data(model, idata_kwargs, data_dict)
+
+    draws2data_dataset = idata_orig.prior_predictive
+
+    # prepare gathering posterior
+    sbc_list = []
+
+    for s in range(model.precision_context.S):
+        draws2data_s = draws2data_dataset.isel(prior_draw=s)
+        data_dict = {
+            **draws2data_s[model.get_obs_vector_names()], # specific to s th synthetic data (obs)
+            **numeric, # driving data (don't use model.numeric which is precision + numeric)
+            **precision
+        }
+        if 'process_noise_scale' in numeric.keys():
+            draws2data_s['process_noise_scale'] = 0.0
+        data2draws_idata_s = data2draws(model, idata_kwargs, data_dict)
+        sbc_list.append(data2draws_idata_s)
+
+    post = xr.concat((data2draws_idata_s.posterior for data2draws_idata_s in sbc_list), dim="prior_draw")
+    post_pred = xr.concat((data2draws_idata_s.posterior_predictive for data2draws_idata_s in sbc_list), dim="prior_draw")
+    loglik = xr.concat((data2draws_idata_s.log_likelihood for data2draws_idata_s in sbc_list), dim="prior_draw")
+    idata_orig.add_groups(posterior=post, posterior_predictive = post_pred, observed_data = draws2data_dataset, log_likelihood = loglik)
+
+    if 'process_noise_scale' in numeric.keys():
+        idata2netcdf4store(f"{get_data_path(model.model_name)}/sbc_{len(setting['est_param_names'])}est_pnoise{numeric['process_noise_scale']}.nc", idata_orig)
+    else:
+        idata2netcdf4store(f"{get_data_path(model.model_name)}/sbc_{len(setting['est_param_names'])}est_pnoise0.nc", idata_orig)
+    plot_qoi(idata_orig, setting, precision, idata_kwargs, model.model_name)
+    #     test_q_lst = ['loglik']
+    #     return diagnose(sbc_idata, test_q_lst)
+    return idata_orig, model
+def update_numeric_obs(model, draws2data_s):
+    """
+    Parameters
+    ----------
+    draws2data_idata: InferenceData with dimension (general: chain, draws, initial_outcome_dim_0, integrated_result_dim_0, integrated_result_dim_1,
+                                                   specific: prey_dim_0, process_noise_dim_0, predator_dim_0, prey_obs_dim_0, predator_obs_dim_0)
+
+    Returns
+    -------
+    model_data2draws: process noise turned-off model
+    numeric_data2draws: draws2data_s added as observed data
+    """
+    matching_vector_names = model.get_matching_vector_names()
+
+    obs_dict = {k: v.values.flatten().reshape((model.N, model.R)) for (k, v) in draws2data_s[matching_vector_names].items()}
+    for matching in matching_vector_names:
+        model.update_numeric({f'{matching}': obs_dict[f'{matching}']})
+
+    return model
+
+def transform_input(vensim, setting, precision, numeric, prior, output_format):
```

Comment:
I'm not sure what this function is trying to achieve. I'm guessing that by changing the format that a user would input settings(from a dict/list structure to function arguments), this function can be eliminated?

Reply:
It adds region subscript to the output_format for hierarchical model. Automating this is top priority.